ci: debezium-postgres failure

The debezium-postgres test has suddenly started failing in CI. Here's a representative example: https://buildkite.com/materialize/tests/builds/35472#01812f87-b52d-402e-ab28-52435366028b

This error only reproduces in CI. I can't reproduce it on my Mac machine, my Linux machine outside of the ci-builder, nor my Linux machine inside of the ci-builder. I can only reproduce it on a CI agent. (Classic.)

The problem occurs with [24-types-enum.td](https://github.com/MaterializeInc/materialize/blob/main/test/debezium/postgres/24-types-enum.td). This test inexplicably causes the Debezium connector to crash (but again, only in CI):

```
debezium_1         | org.apache.kafka.connect.errors.ConnectException: Database connection failed during resolving unknown type
debezium_1         | 	at io.debezium.connector.postgresql.TypeRegistry.resolveUnknownType(TypeRegistry.java:376)
debezium_1         | 	at io.debezium.connector.postgresql.TypeRegistry.get(TypeRegistry.java:212)
debezium_1         | 	at io.debezium.connector.postgresql.connection.PostgresConnection.doReadTableColumn(PostgresConnection.java:570)
debezium_1         | 	at io.debezium.connector.postgresql.connection.PostgresConnection.readColumnForDecoder(PostgresConnection.java:538)
debezium_1         | 	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.getTableColumnsFromDatabase(PgOutputMessageDecoder.java:358)
debezium_1         | 	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.handleRelationMessage(PgOutputMessageDecoder.java:295)
debezium_1         | 	at io.debezium.connector.postgresql.connection.pgoutput.PgOutputMessageDecoder.processNotEmptyMessage(PgOutputMessageDecoder.java:185)
debezium_1         | 	at io.debezium.connector.postgresql.connection.AbstractMessageDecoder.processMessage(AbstractMessageDecoder.java:33)
debezium_1         | 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.deserializeMessages(PostgresReplicationConnection.java:510)
debezium_1         | 	at io.debezium.connector.postgresql.connection.PostgresReplicationConnection$1.readPending(PostgresReplicationConnection.java:502)
debezium_1         | 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.processMessages(PostgresStreamingChangeEventSource.java:207)
debezium_1         | 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.execute(PostgresStreamingChangeEventSource.java:169)
debezium_1         | 	at io.debezium.connector.postgresql.PostgresStreamingChangeEventSource.execute(PostgresStreamingChangeEventSource.java:41)
debezium_1         | 	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:174)
debezium_1         | 	at io.debezium.pipeline.ChangeEventSourceCoordinator.executeChangeEventSources(ChangeEventSourceCoordinator.java:141)
debezium_1         | 	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:109)
debezium_1         | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
debezium_1         | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
debezium_1         | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
debezium_1         | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
debezium_1         | 	at java.base/java.lang.Thread.run(Thread.java:829)
debezium_1         | Caused by: org.postgresql.util.PSQLException: This connection has been closed.
debezium_1         | 	at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:883)
debezium_1         | 	at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:1744)
debezium_1         | 	at org.postgresql.jdbc.PgConnection.prepareStatement(PgConnection.java:430)
debezium_1         | 	at io.debezium.connector.postgresql.TypeRegistry.resolveUnknownType(TypeRegistry.java:370)
debezium_1         | 	... 20 more
debezium_1         | 2022-06-04 16:52:04,915 INFO   Postgres|postgres|streaming  Connection gracefully closed   [io.debezium.jdbc.JdbcConnection]
debezium_1         | 2022-06-04 16:52:04,916 INFO   Postgres|postgres|streaming  Connection gracefully closed   [io.debezium.jdbc.JdbcConnection]
debezium_1         | 2022-06-04 16:52:04,916 INFO   Postgres|postgres|streaming  Finished streaming   [io.debezium.pipeline.ChangeEventSourceCoordinator]
debezium_1         | 2022-06-04 16:52:04,916 INFO   Postgres|postgres|streaming  Connected metrics set to 'false'   [io.debezium.pipeline.ChangeEventSourceCoordinator]
debezium_1         | 2022-06-04 16:52:05,069 INFO   ||  WorkerSourceTask{id=psql-connector-0} Either no records were produced by the task since the last offset commit, or every record has been filtered out by a transformation or dropped due to transformation or conversion errors.   [org.apache.kafka.connect.runtime.WorkerSourceTask]
debezium_1         | 2022-06-04 16:52:05,071 ERROR  ||  WorkerSourceTask{id=psql-connector-0} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted   [org.apache.kafka.connect.runtime.WorkerTask]
debezium_1         | org.apache.kafka.connect.errors.ConnectException: An exception occurred in the change event producer. This connector will be stopped.
```

What's particularly weird is that the SQL query fails because the connection is closed. There is no further information about why in the PostgreSQL logs.

--- a/misc/python/materialize/mzcompose/services.py
+++ b/misc/python/materialize/mzcompose/services.py
@@ -14,7 +14,14 @@ from typing import Dict, List, Optional, Tuple, Union
 from materialize.mzcompose import Service, ServiceConfig
 
 DEFAULT_CONFLUENT_PLATFORM_VERSION = "7.0.3"
-DEFAULT_DEBEZIUM_VERSION = "1.9"
+
+# Be sure to use a `X.Y.Z.Final` tag here; `X.Y` tags refer to the latest
+# minor version in the release series, and minor versions have been known to
+# introduce breakage.
+#
+# Do not upgrade past v1.9.2.Final until debezium/debezium#3570 is resolved.
+DEFAULT_DEBEZIUM_VERSION = "1.9.2.Final"
+
 LINT_DEBEZIUM_VERSIONS = ["1.4", "1.5", "1.6"]
 
 DEFAULT_MZ_VOLUMES = [

